olp-cpp-sdk-authentication/src/TokenProvider.cpp

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (C) 2019-2021 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include <olp/authentication/TokenProvider.h>
+
+#include "AutoRefreshingToken.h"
+#include "TokenEndpoint.h"
+#include "TokenRequest.h"
+#include "TokenResult.h"
+
```

Comment:
Namespace is missing. Nothing should be placed in the global namespace.

Reply:
these files are in the same directory.

Reply:
I know what you mean, but namespace enclosing is a must.